src/consumer.cpp

```diff
@@ -73,9 +73,14 @@ Consumer::~Consumer() {
         rebalance_error_callback_ = nullptr;
         close();
     }
-    catch (const Exception&) {
-        // If close throws just silently ignore until there's some
-        // logging facility (if any)
+    catch (const Exception& ex) {
+        auto& callback = get_configuration().get_log_callback();
+        if (callback) {
+            callback(*this, LOG_ERR, library_name, ex.what());
```

Comment:
Maybe this should actually add some extra information here? I'm not sure the exception message would be enough for the user to realize what happened. Maybe something along the lines of `"Failed to close consumer: <error-message>"`?